### PR TITLE
[Model] add model example GCN-based Anti-Spam

### PR DESCRIPTION
**Pull request by kayzliu**
## Description

add model example GCN-based Anti-Spam from paper [Spam Review Detection with Graph Convolutional Networks](https://arxiv.org/abs/1908.10679.pdf) @zhjwy9343 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/gas/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+----------------------
+- Python 3.7.10
+- PyTorch 1.8.1
+- dgl 0.7.0
+- scikit-learn 0.23.2
+
+Dataset
+---------------------------------------
+The datasets used for edge classification are variants of DGL's built-in fake news datasets. **NOTE**: Same as the original fake news dataset, this variant is for academic use only as well, and commercial use is prohibited. The statistics are summarized as followings:
+
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply examples/pytorch/gas/README.md:23]: Actually user cannot convert the dataset by themselves, because some of the information are missing, but anyway the diagram is still added.

File: examples/pytorch/gas/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Dependencies
+----------------------
+- Python 3.7.10
+- PyTorch 1.8.1
+- dgl 0.7.0
+- scikit-learn 0.23.2
+
+Dataset
+---------------------------------------
+The datasets used for edge classification are variants of DGL's built-in fake news datasets. **NOTE**: Same as the original fake news dataset, this variant is for academic use only as well, and commercial use is prohibited. The statistics are summarized as followings:
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply examples/pytorch/gas/README.md]: Sure.

File: examples/README.md [hunk 3]
```diff
@@ -258,6 +259,9 @@ The folder contains example implementations of selected research papers related
 - <a name='gtn'></a> Yun S, Jeong M, et al. Graph transformer networks. [Paper link](https://arxiv.org/abs/1911.06455).
     - Example code: [OpenHGNN](https://github.com/BUPT-GAMMA/OpenHGNN/tree/main/openhgnn/output/GTN)
     - Tags:  Heterogeneous graph, Graph neural network, Graph structure
+- <a name='gas'></a> Li A, Qin Z, et al. Spam Review Detection with Graph Convolutional Networks. [Paper link](https://arxiv.org/abs/1908.10679).
+    - Example code: [PyTorch](../examples/pytorch/gas)
+    - Tags:  Fraud detection, Heterogeneous graph, Edge classification
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply examples/README.md]: Sure.

File: examples/pytorch/gas/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+If the mini-batch training is required to run on a GPU, run
+```
+python main_sampling.py --gpu 0
+```
+
+Performance
+-------------------------
+|Dataset               | Xianyu (reported) | Fake News Politifact | Fake News Gossipcop |
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply examples/pytorch/gas/README.md]: Sure, but I think use the term of Xianyu Graph, which is used in the paper, is better.

File: examples/pytorch/gas/dataloader.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        assert 0 <= train_size + val_size <= 1, \
+            "The sum of valid training set size and validation set size " \
+            "must between 0 and 1 (inclusive)."
+
+        num_edges = graph.num_edges(etype='forward')
+        index = np.arange(num_edges)
+
+        index = np.random.RandomState(seed).permutation(index)
+        train_idx = index[:int(train_size * num_edges)]
+        val_idx = index[len(index) - int(val_size * num_edges):]
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply examples/pytorch/gas/dataloader.py]: Sure.

File: examples/pytorch/gas/model.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        nv_dim = int(v_out_dim / 2)
+
+        self.W_u = nn.Linear(v_in_dim + e_in_dim, nu_dim)
+        self.W_v = nn.Linear(u_in_dim + e_in_dim, nv_dim)
+
+        self.Vu = nn.Linear(u_in_dim, u_out_dim - nu_dim)
+        self.Vv = nn.Linear(v_in_dim, v_out_dim - nv_dim)
+
+    def forward(self, g, e_feat, u_feat, v_feat):
+        g.nodes['u'].data['h'] = u_feat
```
[the remark this answers is not in the document]
Comment (kayzliu) [reply examples/pytorch/gas/model.py]: Thanks for your suggestion, and it's added in the latest commit.

**Review by zhjwy9343 — APPROVED** [review]
